### PR TITLE
Indicate the TTL of logs in Elasticsearch reporting on the Lifecycle Policy of the `logs-apm.app-*` index template

### PR DESCRIPTION
**Pull request by cyrille-leclerc**
Implementation of 
- https://github.com/jenkinsci/opentelemetry-plugin/issues/334
 
Indicate the TTL of logs in Elasticsearch reporting on the Lifecycle Policy of the `logs-apm.app-*` index template.

⚠️ we hard code the name of Lifecycle Policy instead of retrieving it from the index teamplate waiting for 
* https://github.com/jenkinsci/opentelemetry-plugin/issues/336

I had to switch to the elasticsearch-java:8.0.0 client to be able to query the IndexTemplate.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

Example report of a successful "Validate Elasticsearch configuration":
```
Connected to Elasticsearch https://*******.europe-west1.gcp.cloud.es.io:9243
Index Template 'logs-apm.app' found
Index Lifecycle Policy 'logs-apm.app_logs-default_policy' found. Retention strategy: hot[rollover[maxAge=30d, maxSize=50gb]], warm [phase not defined], cold [phase not defined], delete[delete[min_age=10d]]
```

image

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticsearchLogStorageRetriever-v7.txt [hunk 1]
```diff
@@ -0,0 +1,303 @@
+/*
```
Comment (kuisathaverat) [inline src/main/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticsearchLogStorageRetriever-v7.txt:1]: a txt file? looks like a leftover

File: src/main/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticsearchLogsQueryContext-v7.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/*
```
Comment (kuisathaverat) [inline src/main/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticsearchLogsQueryContext-v7.txt:1]: ditto

File: src/main/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticsearchLogStorageRetriever-v7.txt (hunk 1 above)
Comment (cyrille-leclerc) [reply src/main/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticsearchLogStorageRetriever-v7.txt:1]: kept here to be prepared to rollback or compare if needed.

File: src/main/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticsearchLogsQueryContext-v7.txt (hunk 2 above)
Comment (cyrille-leclerc) [reply src/main/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticsearchLogsQueryContext-v7.txt:1]: kept here to be prepared to rollback or compare if needed.

**Pull request merged**